backend/molgenis-emx2-sql/src/main/java/org/molgenis/emx2/sql/MetadataUtils.java

```diff
@@ -677,11 +679,37 @@ public static User loadUserMetadata(SqlDatabase db, String userName) {
     if (userRecord != null) {
       User result = new User(db, userName);
       result.setSettings(userRecord.get(SETTINGS, Map.class));
+      Map<String, String> roles = userRecord.get("roles", Map.class);
+      if (roles != null) {
+        result.setRoles(roles);
+      }
       return result;
     }
     return null;
   }
 
+  public static List<Member> loadUserRoles(SqlDatabase db) {
+    List<Member> members = new ArrayList<>();
+    String roleFilter = Constants.MG_ROLE_PREFIX;
+    String userFilter = Constants.MG_USER_PREFIX;
+    List<Record> result =
```

Comment:
might need txAsAdmin wrapper (I didn't look yet to context of the call,  maybe that is already admin)

Reply:
It works, so I guess it's not needed. Probably since this screen is admin only, you're always already logged in as admin.